ENH - Scalable colour system for theme

trallard opened the pull request:
Closes #1079 
Closes #1052 

As discussed in gh-1079 this PR will focus on adopting the new colour scheme and _not_ any substantial style changes (these will come in a separate PR). Though I might need to adjust some styles/variables as part of this work, I will do my best to keep you all aligned and update the docs as needed.

---

🏷️  **Edited: I am taking this out of WIP to encourage reviews, comments and questions.**

Things I am still not super happy with or need work on (I will check the items as I work on them):

- [x] Scrollbars - I am not super happy with the current colours; they are either too dark or too light and draw a lot of attention
- [x] Buttons - these inherit from `sphinx design` - I have changed the formula to decide on the text colour (using [WCAG luminance](https://www.w3.org/TR/WCAG/#dfn-relative-luminance) and [contrast ratio formulas](https://www.w3.org/TR/WCAG/#dfn-contrast-ratio) but for some reason the mixins in `_sphinx_design.scss` are giving me trouble so will debug here)
- [x] Badges - same issue as buttons
- [x] Buttons - fix hover colour (separate PR for buttons)
- [x] Fix dropdown colours (sphinx design - little tweaks)
- [x] Little tweaks needed on the admonition colour `mixin`
- [x] Shadows improvements (nothing major, simple tweaks here and there)
- [x] Adjust banner link colours - light theme only
- [x] Documentation:
	- [x] Update accessibility section (user)
	- [x] Update accessibility section (developer)
	- [ ] Update theming and customisation sections  	

Sans those little nags and whatever we encounter during the review, this seems to be in a decent place to start reviewing and hopefully merge soonish 🚀 

Additional notes:

-  Right now, the announcement bars use the `info` colour for the bg: [in the proposal Figma file](https://www.figma.com/file/rUrrHGhUBBIAAjQ82x6pz9/PyData-Design-system---proposal-for-implementation-(2)?node-id=1238%3A983&t=ckaCBlZPqzjDQdNR-1) I had used primary/secondary/accent for these. I changed the colour to `secondary` in this PR, so I have the following questions
 	- Do we want to keep `info` as the default colour? I am neutral about this
 	- Shall we have 2-3 slight variants so folks can choose the colour from, let's say, `info, primary, secondary, accent`?
 	
For quick checks here is the link [to the Read the docs PR preview](https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174)

drammock commented on src/pydata_sphinx_theme/assets/styles/variables/_color.scss:
```diff
@@ -5,97 +5,160 @@
+      // gray-800
     ),
   // surface: object set above the background (without shadows)
   // Uses JupyterLab cell background colors in light/dark theme
```
comment on this line is probably obsolete since we're about to change the colors.

drammock commented on src/pydata_sphinx_theme/assets/styles/variables/_color.scss:
```diff
@@ -5,97 +5,160 @@
-    // White - 50
+    "light": rgb(34, 40, 50),
+    // gray-800
+    // "dark": #e5e7ea
```
cruft line?

drammock reviewed (commented):
I know this is WIP but I was curious 🙂 just two little comments of things I noticed along the way

trallard commented:
Thanks - this is due to change plenty as I work towards the final bits. But an early review is always welcome.

trallard commented:
🤔  As I have been working on the theme today, I started getting a weird error while trying to use `nox -s docs-live` while it was working well before, now it fails with the following error:

```
File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/jupyter_cache/cache/main.py", line 300, in get_cache_bundle
    raise KeyError(f"Notebook file does not exist for cache record PK: {pk}")
KeyError: 'Notebook file does not exist for cache record PK: 1'

Exception occurred:
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/jupyter_cache/cache/main.py", line 300, in get_cache_bundle
    raise KeyError(f"Notebook file does not exist for cache record PK: {pk}")
KeyError: 'Notebook file does not exist for cache record PK: 1'
The traceback has been saved in /var/folders/6p/062gm41556s5p63x1555cjzw0000gn/T/sphinx-err-i8usg03k.log, if you want to report the issue to the developers.
```

About my setup:

```
Sphinx version: 5.3.0
# Python version: 3.9.16 (CPython)
# Docutils version: 0.19 
# Jinja2 version: 3.1.2
```

Extended log-in here if needed

<details>
# Sphinx version: 5.3.0
# Python version: 3.9.16 (CPython)
# Docutils version: 0.19 
# Jinja2 version: 3.1.2
# Last messages:
#   reading sources... [ 50%] examples/kitchen-sink/really-long
#   reading sources... [ 51%] examples/kitchen-sink/structure
#   reading sources... [ 52%] examples/kitchen-sink/tables
#   reading sources... [ 53%] examples/kitchen-sink/typography
#   reading sources... [ 54%] examples/mult_headers
#   reading sources... [ 56%] examples/no-sidebar
#   reading sources... [ 57%] examples/persistent-search-field
#   reading sources... [ 58%] examples/pydata
#   /Users/tania/Documents/github/patches/pydata-sphinx-theme/docs/examples/pydata.md: Using cached notebook: ID=1 [mystnb]
#   sphinx-sitemap: No pages generated for sitemap.xml
# Loaded extensions:
#   sphinx.ext.mathjax (5.3.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/ext/mathjax.py
#   sphinxcontrib.applehelp (1.0.4) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinxcontrib/applehelp/__init__.py
#   sphinxcontrib.devhelp (1.0.2) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinxcontrib/devhelp/__init__.py
#   sphinxcontrib.htmlhelp (2.0.1) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinxcontrib/htmlhelp/__init__.py
#   sphinxcontrib.serializinghtml (1.1.5) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinxcontrib/serializinghtml/__init__.py
#   sphinxcontrib.qthelp (1.0.3) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinxcontrib/qthelp/__init__.py
#   alabaster (0.7.13) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/alabaster/__init__.py
#   sphinx.ext.autodoc.preserve_defaults (5.3.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/ext/autodoc/preserve_defaults.py
#   sphinx.ext.autodoc.type_comment (5.3.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/ext/autodoc/type_comment.py
#   sphinx.ext.autodoc.typehints (5.3.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/ext/autodoc/typehints.py
#   sphinx.ext.autodoc (5.3.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py
#   sphinx.ext.autosummary (5.3.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/ext/autosummary/__init__.py
#   sphinx.ext.todo (5.3.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/ext/todo.py
#   sphinx.ext.viewcode (5.3.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/ext/viewcode.py
#   sphinxext.rediraffe (unknown version) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinxext/rediraffe.py
#   sphinx_design (0.3.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx_design/__init__.py
#   sphinx_copybutton (0.5.1) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx_copybutton/__init__.py
#   ablog (0.11.0rc2) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/ablog/__init__.py
#   jupyter_sphinx (0.4.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/jupyter_sphinx/__init__.py
#   matplotlib.sphinxext.plot_directive (3.7.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/matplotlib/sphinxext/plot_directive.py
#   myst_nb (0.17.1) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/myst_nb/__init__.py
#   sphinxcontrib.youtube (unknown version) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinxcontrib/youtube/__init__.py
#   numpydoc (1.5.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/numpydoc/__init__.py
#   sphinx_togglebutton (0.3.2) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx_togglebutton/__init__.py
#   sphinx_sitemap (2.5.0) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx_sitemap/__init__.py
#   pydata_sphinx_theme (unknown version) from /Users/tania/Documents/github/patches/pydata-sphinx-theme/src/pydata_sphinx_theme/__init__.py
Traceback (most recent call last):
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/cmd/build.py", line 281, in build_main
    app.build(args.force_all, args.filenames)
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/application.py", line 343, in build
    self.builder.build_all()
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 269, in build_all
    self.build(None, summary=__('all source files'), method='all')
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 326, in build
    updated_docnames = set(self.read())
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 433, in read
    self._read_serial(docnames)
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 454, in _read_serial
    self.read_doc(docname)
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 510, in read_doc
    publisher.publish()
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/docutils/core.py", line 224, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/sphinx/io.py", line 104, in read
    self.parse()
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/docutils/readers/__init__.py", line 76, in parse
    self.parser.parse(self.input, document)
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/myst_nb/sphinx_.py", line 150, in parse
    with create_client(
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/myst_nb/core/execute/base.py", line 83, in __enter__
    self.start_client()
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/myst_nb/core/execute/cache.py", line 36, in start_client
    _, self._notebook = cache.merge_match_into_notebook(self.notebook)
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/jupyter_cache/cache/main.py", line 357, in merge_match_into_notebook
    cache_nb = self.get_cache_bundle(pk).nb
  File "/Users/tania/Documents/github/patches/pydata-sphinx-theme/.nox/docs-live/lib/python3.9/site-packages/jupyter_cache/cache/main.py", line 300, in get_cache_bundle
    raise KeyError(f"Notebook file does not exist for cache record PK: {pk}")
KeyError: 'Notebook file does not exist for cache record PK: 1'

</details>

I have removed my `.nox` directory a couple of times and had no luck - I might try to manually setup my env tomorrow, as it would be best if I could build and serve the docs locally to finish the PR. 
The docs, however, seem to be building in Read the docs see https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174/ 

Is anyone has any pointers here it would be grand

drammock commented:
> Notebook file does not exist for cache record PK

This is a new one, never seen it before. It's coming from jupyter-cache via Myst-NB, which means @choldgraf is probably the most knowledgable here. Are you still stuck there @trallard ?

trallard commented:
I resourced to start from scratch - new env, clone repo again and it's the only way I managed to get rid of this.

I am not sure how this happened tbh as it only appeared after I merged `main` back into my branch.

choldgraf commented:
Re: the cache, I think you need to delete the jupyter_cache folder in the build directory, that's where all the cached notebooks go. Then it should review from scratch

trallard commented:
I tried that too but the only workaround was starting from scratch.

12rambau commented:
> I tried that too but the only workaround was starting from scratch.

I struggled to build the documentation for at least 1 year on my personnal computer because of sphinx-jupyter until ... I changed it. So in the end starting from scratch is a small price to pay to get it working ;-)

choldgraf commented:
This is shaping up really nicely! A few quick things I noticed:


### instructions for old colors?

I think it'd be helpful if we included a dedicated section for "how to recreate most of the old visual style" that told people which CSS variables and color values to use

### contrast for announcement

I suspect the contrast for links in the announcement banner is too low. What if instead we allowed announcement banners themselves to be clickable as a link (w a config param) and then discouraged the use of links in the banners?

![Screenshot_20230224-143912.png](https://user-images.githubusercontent.com/1839645/221193066-7c366b2b-91f5-4c14-97fb-d81adc6e4c02.png)

### inline code padding

The top padding seems a bit lopsided for our inline code (at least in the toctree where I noticed it)

![Screenshot_20230224-143900.png](https://user-images.githubusercontent.com/1839645/221193402-193f4d8d-2e15-40f4-a5a8-25536b70a4fc.png)

trallard commented:
- agree: I am planning to give the customisation docs a good update so definitely thinking of compatibility/updates
- announcement banner: yes I noticed this - I can always do some tweaks to the colours but also making clickable would be a good addition 
- will double check the code padding, I don't think I touched this at all but I might have accidentally moved something

12rambau commented:
Just passing by to say the colors look amazing ! I'm a huge fan of what you did with the dark theme. 

I have 2 comments: 
- maybe take advantage of this PR to unify the behavior of the theme switcher btn (this is the only one in the navbar with hover fillcolor). I don't see the sun inside in the light theme. 
- update the css/doc used in the to build the customized admonition: https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174/user_guide/extending.html as it doesn't work anymore

trallard commented:
Phew! I finally had some time to add more fixes (I updated the initial comment to reflect the status)

@12rambau I fixed the custom admonitions issue, so this is working again. Also I think I fixed the issues with the theme switcher (but let me know if I missed it or could be tweaked).

I also updated the accessibility sections to reflect the changes. I reviewed several components to ensure we use the new mixing I added to compute text/background colour contrast per WCAG.

There are some "little" issues here and there, but I think I have finally sorted all the big chunks and tricky bits. I have left `TODO` comments for me to go back and fix some things, but this PR is, I'd say, 80% ready. So a check and comments would be greatly appreciated.

12rambau commented:
@trallard as this work is ongoing what about switching from PR to draft ?

trallard commented:
Done! Had taken it out of draft to get the first round of reviews but considering I still am fixing some bits I have marked it as draft again

drammock commented:
This looks really nice!  Here are some small things I noticed:

[attention admonition isn't getting a color](https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174/examples/kitchen-sink/admonitions.html):

![Screenshot 2023-03-22 at 16-32-00 Admonitions — PyData Theme 0 13 2dev0 documentation](https://user-images.githubusercontent.com/1810515/227043322-dad87a35-a5fd-4501-bb6a-674e4a14c9c4.png)

[block quotes and rubrics have less contrast than I expected](https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174/examples/kitchen-sink/blocks.html):

![Screenshot 2023-03-22 at 16-34-42 Blocks — PyData Theme 0 13 2dev0 documentation](https://user-images.githubusercontent.com/1810515/227043851-cada659d-ac86-47ae-8142-90bb6139223b.png)

[...which is especially bad if they contain links](https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174/examples/kitchen-sink/generic.html#rubric):

![Screenshot 2023-03-22 at 16-38-05 Generic items — PyData Theme 0 13 2dev0 documentation](https://user-images.githubusercontent.com/1810515/227044498-f40cfa2d-7990-4f39-84ec-1f32bc70d85b.png)

[Something weird is going on with GUI labels](https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174/examples/kitchen-sink/generic.html#gui-labels):

![Screenshot 2023-03-22 at 16-37-06 Generic items — PyData Theme 0 13 2dev0 documentation](https://user-images.githubusercontent.com/1810515/227044248-2589699e-679e-4d6e-9d1e-bb6527dc0af5.png)

Some (but not all) code boxes on [this page](https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174/examples/pydata.html#pandas) have a green left edge; is that supposed to happen?

12rambau commented:
> Some (but not all) code boxes on [this page](https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174/examples/pydata.html#pandas) have a green left edge; is that supposed to happen?

The green one are a mark from mystnb to show cells that have been executed. THe only one not green is jupyter-sphinx as it's not executed the same way. In https://github.com/pydata/pydata-sphinx-theme/pull/1023 I'd like to move it with jupyterlite in an "execution" page to avoid confusion.

trallard commented on src/pydata_sphinx_theme/assets/styles/extensions/_sphinx_design.scss:
```diff
@@ -50,32 +41,63 @@ $sd-semantic-color-names: (
+  @debug "created color 🌱";
+  // we are now using a11y-combination to calculate the text color - this is based
+  // on the WCAG color contrast guidelines
+  --pst-color-#{$name}-text: #{a11y-combination($value)};
```
@choldgraf I made a drastic change on how to assign/define text colour (to meet WCAG contrast criteria), but since the method here was an adaptation of what sphinx design uses I am thinking of adding this new approach on sphinx design too

trallard commented:
Ok, I am taking this out of draft as I did a last check, and the colour contrast issues are gone, and I have unified all the colours throughout. 
A check from y'all would be helpful.

The only remaining item for me to do should be reviewing and [updating the customisation/theming docs sections](https://github.com/pydata/pydata-sphinx-theme/pull/1174#issue-1582245220). But I want to ensure the implementation works appropriately before updating this section.

There might be some minor colour issues I might have missed but those can be ironed out as we do more accessibility audits and improve our tests.

12rambau reviewed (commented):
@trallard congratulation to put everything together, that's a huge piece of work and I really love what you did with the color scheme espacially the dark theme. I noticed few things that may or may not require your attention before merging (if you want to take care of them in a follow-up PR):

- in https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174/examples/persistent-search-field.html the magnifying glass icon is outside of the search field
- I'm not 100% sure about the purple in Todo admonitions (it's usually greyish in other theme but maybe it's as teal I just need to get used to it)
- the "remove search result" button (https://pydata-sphinx-theme--1174.org.readthedocs.build/en/1174/examples/kitchen-sink/index.html#kitchen-sink) at the top of the page seems stands out too much (too, big and not very handy) compared to the fine tuned components you created

12rambau commented:
> we already have such a colour combination elsewhere

If the color combination is taken by the default admonition then don't change anything, it's better if things look a bit different

trallard commented:
I could try and change the semantic colours to the next shade in light mode. 
That should not affect the legibility or accessibility conformance and I can make this change pretty easily. Same with the border colour for the blocks.

The search results one - I did not change this at all as when this was discussed there was some sort of an agreement to keep as is. That said I did have some yellow shades in the very original research phases so I can try and change the current one to fix this contrast issue.

12rambau commented:
for the target color, I rememember it's rather complicated to get something prominent enough to stand out against anything (link, admonition, code, regular text, titles... etc) I would merge it as it is and deal with it in a follow-up PR.

trallard commented:
As I am adding the finishing touches and updates to docs I realised we'd need to update the screenshot in the Readme. Does anyone know how to create this type of half and half images?

12rambau commented:
I did them with powerpoint. take 2 screenshots of the same page in each theme and cut one of them with a triangle then combine and save as image. The shadow is a byproduct of Mac screeenshots but I'm not sure it's necessary.

trallard commented:
@drammock, thanks for the feedback on the admonitions, I fixed the colours on the light theme, and I agree these look much in line with the dark theme now

![pyData Sphinx theme admonitions page](https://github.com/pydata/pydata-sphinx-theme/assets/23552331/41861984-8a09-40fd-a2fb-461820b6aea0)

Also, I replaced the inline code borders in the dark theme for a less bright shade of gray so hopefully, this will be less disruptive/jarring. 

![pydata sphinx theme pandas example](https://github.com/pydata/pydata-sphinx-theme/assets/23552331/5041f5ce-6faa-408c-a344-a667250e76fc)

As for the `highlight target` colour - as @12rambau mentioned, it is pretty hard to get colours with good contrast against all possible text colours in the theme. I tried many combinations, and this is the best I could get to:

![Read_the_Docs_functionality_—_PyData_Theme_0_13_4dev0_documentation](https://github.com/pydata/pydata-sphinx-theme/assets/23552331/2fb98933-728f-4716-a56c-daad1f43a31f)

![Read_the_Docs_functionality_—_PyData_Theme_0_13_4dev0_documentation](https://github.com/pydata/pydata-sphinx-theme/assets/23552331/81941a3e-2755-414f-896a-6393550ce836)

The target colour `#675c04` for the dark theme meets WCAG AA contrast against links, text, and background (3:1)
the one on the light theme is way trickier to get the right contrast against background and text colours as these are opposite (white vs. almost dark), so the best I could do was set this to `#f3cf95` which is also slightly shifted towards red vs. the dark target theme (IMHO not a big deal). But overall, these have better contrast and legibility than the ones we had.

trallard commented on .jupyterlite.doit.db:
Oops 😬 unintended add. Will add it to the .gitignore

choldgraf commented:
Just took a quick look, I am +1 on merging. I think it's a clear improvement, and we will get more feedback from ourselves and others once we put it in front of people so I think it's best to merge and iterate rather than trying to get it perfect this time. If y'all want to finish a few things first though that's fine too.

trallard commented on docs/user_guide/accessibility.rst:
```diff
@@ -60,9 +66,9 @@ identify if the content is in a language the reader understands.
 .. Hint::
 
-    Specifying a ``language`` will propagate to the top-level `html` tag.
+    Specifying a ``language`` will propagate to the top-level `HTML` tag.
```
I keep getting this wrong moving between rst and md 🤦🏻‍♀️

trallard commented on docs/user_guide/styling.rst:
```diff
@@ -114,17 +117,18 @@ Here is an overview of the colors available in the theme (change theme mode to s
-      <span class="sd-sphinx-override sd-badge pst-badge pst-on-surface">on-surface</span>
-      <span class="sd-sphinx-override sd-badge pst-badge pst-target">target</span>
+      <span class="sd-sphinx-override sd-badge pst-badge pst-on-surface sd-bg-text-primary">on-surface</span>
+      <span class="sd-sphinx-override sd-badge pst-badge pst-target sd-bg-text-primary">target</span>
```
Good catch I forgot to update this after updating the target colour as in the previous combination the shades were inverted

trallard commented on src/pydata_sphinx_theme/assets/styles/content/_admonitions.scss:
```diff
@@ -161,7 +162,7 @@ div.admonition,
     > .admonition-title {
       &:before {
-        background-color: var(--pst-color-attention);
+        background-color: var(--pst-color-warning-bg);
```
good catch! fixing now

trallard commented on src/pydata_sphinx_theme/assets/styles/components/_searchbox.scss:
```diff
@@ -24,8 +24,7 @@ div#searchbox {
       background-color: var(--pst-color-primary);
-      // Button text is always white with Sphinx Design buttons
-      color: white;
+      color: var(--pst-color-primary-text);
```
Thanks for flagging this! Honestly this is something I completely missed as I created some of these variables on the fly.

> where the intended page element (either -bg or -text) comes last.
This makes sense, will make a pass to fix these

trallard replied:
On a quick check some variables like `text-base` seem to have already been there when I made the change (see in [colour.scss](https://github.com/trallard/pydata-sphinx-theme/blob/e5830a28fb760cc15688fcfa4dc4e8a4600225d7/src/pydata_sphinx_theme/assets/styles/variables/_color.scss#L168)) but I can go ahead and unify these

trallard replied:
I just noticed that since [sphinx_design.scss creates its variables to work with sphinx design components](https://github.com/trallard/pydata-sphinx-theme/blob/559b1f789df45b1ce7783f3d02e64840ce3d14b8/src/pydata_sphinx_theme/assets/styles/extensions/_sphinx_design.scss#L71) this effectively overrides what would then be `--pst-color-muted-text` which now makes me believe this is the reason why  `--pst-color-text-muted` was defined in the first place

drammock commented:
hooray!!! in it goes, thanks a million @trallard this is a big big improvement that I'm really pleased with.

choldgraf commented:
Woohoo! 🎉

trallard commented:
🎉  Thank you folks it was a long and complex PR, but y'all were more than supportive and open throughout all these months!

12rambau commented:
should we make a  feature release so that people give us feedback ?

trallard commented:
It might be helpful, but could we wait until #1353 is merged? With those changes we fix a lot of accessibility issues cropping in our tests

12rambau commented:
I added the block release tag to make sure I don't forget

Gouvernathor commented:
So, apparently _this_ is the place where the dark theme's background color was changed to bluish 😥
I don't see any discussion on anything blue, and discussions related to the background either mention the light background or pertain to the color of other things relate to the background color having already been changed.

I find the former, very dark grey color was much better in the style and impression it gave out to the rest of the page, and that this particular change is quite bad.
I also though the same about the link colors that were changed (maybe not in this PR), until I learned it was related to accessibility, which makes it acceptable.
However, in the case of the dark theme's background, I don't see it. Since the font color is white, the contrast between the font and te background _decreases_ when you change it from almost pitch-black to bluish, so how can that be supported by an accessibility rationale ?

I guess it may need an issue to start repealing this particular change (I'm not opposed to all the changes in this PR of course), but I would like to learn the rationale for it to understand the context first.

12rambau commented:
As mentioned in the first comment of this fantastic PR (thank you again @trallard for your commitment) these modifications were done after a long conversation in #1079 I don't remember all the threads but no color was changed without a good justification from the a11y experts. 
From a personal POV although I wasn't a fan of the teal as primary color I love the final render. 

Note that if you want, you can still revert the color to what they were in your custom.css.
